wazo_sysconfd/plugins/commonconf/services.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2010-2022 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import logging
+import subprocess
+import traceback
+from wazo_sysconfd import exceptions
+
+logger = logging.getLogger('wazo_sysconfd.modules.commonconf')
+
+commonconf_config = dict()
```

Comment:
I would prefer not to use global variables if we can avoid it. The previous implementation did not require a global variable.

Reply:
The reason why I changed this is because if we keep the old format, we would have to pass the object like the following:
```python
class Plugin:
    def load(self, dependencies: dict):
        options = dependencies['config']

        commonconf = CommonConf()
        commonconf.safe_init(options)
        api = dependencies['api']
        api.include_router(router, dependencies=[Depends(lambda: commonconf)])
```
This will cause the following error when we call `post` and `get` methods within `http.py`:
```
(ERROR) (STDERR): AttributeError: 'CommonConf' object has no attribute 'generate_cmd'
```
The error is due to the fact that for some reason, `fastapi` instead of passing the object that we initiated in plugin, creates another object instead.

Reply:
I understand your reasoning, thanks for explaining. In the other branches, we have added a small cache function to return the same instance each time. I have added that cache in this PR as well, to gain some time :)